<!DOCTYPE html>
<html lang="en">

<head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="content-type" content="text/html;charset=utf-8">
    <title>Wee JunJie</title>
    <meta name="author" content="Wee JunJie">
    <link rel="stylesheet" href="css/latex_style.css">
    <link rel="stylesheet" media="screen" href="css/style.css">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons/css/academicons.min.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0/css/all.min.css">
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-54900133-1"></script>
    <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-54900133-1');
    </script>
</head>

<body>

    <!-- particles.js container -->
    <div id="container">
        <div id="particles-js"></div>
        <div class="container">
            <div id="profpicdiv"></div>
                <div id="contacts">
                <table class="imgtable">
                    <tbody>
                        <tr>
                            <td align="left">
                                <p>
                                <h1>JunJie Wee</h1>
                                <div class="icon-bar" align="left">
                                    <a href="https://scholar.google.com.sg/citations?hl=en&user=aY7AQeIAAAAJ"
                                        target="_blank"><i class="ai ai-google-scholar-square ai-2x"></i></a>
                                    <a href="https://github.com/ExpectozJJ/" target="_blank"><i
                                            class="fa-brands fa-github-square fa-2x"></i></a>
                                    <a href="https://orcid.org/my-orcid?orcid=0000-0001-8444-3252" target="_blank"><i
                                            class="ai ai-orcid-square ai-2x"></i></a>
                                    <a href="https://www.researchgate.net/profile/Junjie-Wee/" target="_blank"><i
                                            class="ai ai-researchgate-square fa-2x"></i></a>
                                            <a href="https://publons.com/researcher/4297976/junjie-wee" target="_blank"><i
                                                class="ai ai-publons-square fa-2x"></i></a>
                                </div>
                                <br>

                                <li id="email" style="font-size: 14px;margin-left:0;list-style-type: none;font-family:Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;"> <i class="fas fa-envelope"></i> weej0019[at]e.ntu.edu.sg</li><br></ul>
                                <ul>
                                <li itemscope id="office" style="font-size: 14px;margin-left:0;list-style-type: none;"> 
                                  <i class="fas fa-briefcase"></i> SPMS-MAS-04-05</li>
                                  <li itemscope id="office" style="font-size: 14px;margin-left:9;list-style-type: none;"> 
                                    Division of Mathematical Sciences</li>
                                    <li itemscope id="office" style="font-size: 14px;margin-left:9;list-style-type: none;"> 
                                      School of Physical & Mathematical Sciences</li>
                                      <li itemscope id="office" style="font-size: 14px;margin-left:9;list-style-type: none;"> 
                                        Nanyang Technological University</li>
                                        <li itemscope id="office" style="font-size: 14px;margin-left:9;list-style-type: none;"> 
                                          21 Nanyang Link, Singapore 637371</li>
                                </ul>
                            </td>
                        </tr>
                    </tbody>
                </table>

            </div>

            <br><br>
            <div id="menu" class="">
                <ul id="ulmenu" class="">
                  <li><a href="index.html" id="ho" class="" style="font-size: 18px;">Home</a></li>
                <li><a href="publications.html" id="pu" class="on" style="font-size: 18px;">Publications</a></li>
                <li><a href="talks.html" id="ta" class="" style="font-size: 18px;">Talks/Conferences</a></li>
                <li><a href="teachings.html" id="te" class="" style="font-size: 18px;">Teaching</a></li>
            </ul>
            <a class="dropdown" style="float:right;">
                <button onclick="myFunction()" class="dropbtn">☰</button>
                <div id="myDropdown" class="dropdown-content" align="left">
                    <a href="index.html">Home</a>
                    <a href="publications.html">Publications</a>
                    <a href="talks.html">Talks/Conferences</a>
                    <a href="teachings.html">Teaching</a>
                </div>
            </a>
            </div>

            <div id="more">
                <div id="splash" style="display: none;" class="gone">
                    <p>To view this webpage, <b>javascript</b> must be <b>enabled</b>.</p>
                </div>
                <br><br><br>

                <h2>Publications <button class="button" id="ec" onclick="openAll()">+</button></h2>
                <p style="font-size: 14px">Click the publications to view abstract. Expand/collapse all abstracts by clicking button above.</p><br>
                
                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details>
                            <summary>
                            [12] <strong>Wee JunJie</strong>, Ginestra Bianconi, Xia Kelin. Persistent Dirac for molecular representation. (2023). arXiv preprint arXiv:2302.02386.</summary><br>
                                <p>
                                    Molecular representations are of fundamental importance for the modeling and analysis of molecular systems. Representation models and in general approaches based on topological data analysis (TDA) have demonstrated great success in various steps of drug design and materials discovery. Here we develop a mathematically rigorous computational framework for molecular representation based on the persistent Dirac operator. The properties of the spectrum of the discrete weighted and unweighted Dirac matrices are systemically discussed and used to demonstrate the geometric and topological properties of both non-homology and homology eigenvectors of real molecular structures. This allows us to asses the influence of weighting schemes on the information encoded in the Dirac eigenspectrum. A series of physical persistent attributes, which characterize the spectrum of the Dirac matrices across a filtration, are proposed and used as efficient molecular fingerprints. Finally, our persistent Dirac-based model is used for clustering molecular configurations from nine types of organic-inorganic halide perovskites. We found that our model can cluster the structures very well, demonstrating the representation and featurization power of the current approach.
                                </p>
                                </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://arxiv.org/abs/2302.02386" target="_blank" rel="noopener">
                                DOI
                                </a>
                            </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details>
                            <summary>
                            [11] Choo Hou Yee, <strong>Wee JunJie</strong>, Shen Cong, Xia Kelin. Fingerprint-Enhanced Graph Attention Network (FinGAT) Model for Antibiotic Discovery. <span style="font-style: italic;">Journal of Chemical Information and Modeling</span> (2023).</summary><br>
                                <p>
                                    Artificial Intelligence (AI) techniques are of great potential to fundamentally change antibiotic discovery industries. Efficient and effective molecular featurization is key to all highly accurate learning models for antibiotic discovery. In this paper, we propose a fingerprint-enhanced graph attention network (FinGAT) model by the combination of sequence-based 2D fingerprints and structure-based graph representation. In our feature learning process, sequence information is transformed into a fingerprint vector, and structural information is encoded through a GAT module into another vector. These two vectors are concatenated and input into a multilayer perceptron (MLP) for antibiotic activity classification. Our model is extensively tested and compared with existing models. It has been found that our FinGAT can outperform various state-of-the-art GNN models in antibiotic discovery.
                                </p>
                                </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1021/acs.jcim.3c00045" target="_blank" rel="noopener">
                                DOI
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/FinGAT" target="_blank" rel="noopener">
                                    Code
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1021/acs.jcim.3c00045" data-hide-zero-citations="true" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details ><summary>
                            [10] <strong>Wee JunJie</strong>, Le Hai Khoi. On Korenblum Constants for some weighted function spaces. Journal of 
                            Mathematics and Mathematical Sciences, 2(8). (2023).
                        </summary><br>
                                <p> 	In this paper, we survey the results on the Korenblum Maximum Principle for some weighted function spaces. Progress and results discussed include the upper bounds and lower bounds of Korenblum constants, as well as the failure of the principle for weighted Bergman spaces, weighted Hardy spaces, weighted Bloch spaces, weighted Fock spaces, and mixed norm spaces. Existing and new open questions are provided.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://science.thanglong.edu.vn/index.php/volc/article/view/89" target="_blank" rel="noopener">
                                DOI
                                </a>
                        </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [9] Xia Kelin, Liu Xiang, <strong>Wee JunJie</strong>. Persistent Homology for RNA Data Analysis. In: Filipek, S. (eds) Homology Modeling. Methods in Molecular Biology, (2023). vol 2627. Humana, New York, NY. <strong>(Book Chapter)</strong> 
                        </summary><br>
                                <p> 	Molecular representations are of great importance for machine learning models in RNA data analysis. Essentially, efficient molecular descriptors or fingerprints that characterize the intrinsic structural and interactional information of RNAs can significantly boost the performance of all learning modeling. In this paper, we introduce two persistent models, including persistent homology and persistent spectral, for RNA structure and interaction representations and their applications in RNA data analysis. Different from traditional geometric and graph representations, persistent homology is built on simplicial complex, which is a generalization of graph models to higher-dimensional situations. Hypergraph is a further generalization of simplicial complexes and hypergraph-based embedded persistent homology has been proposed recently. Moreover, persistent spectral models, which combine filtration process with spectral models, including spectral graph, spectral simplicial complex, and spectral hypergraph, are proposed for molecular representation. The persistent attributes for RNAs can be obtained from these two persistent models and further combined with machine learning models for RNA structure, flexibility, dynamics, and function analysis.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1007/978-1-0716-2974-1_12" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a><span class="__dimensions_badge_embed__" data-doi="10.1007/978-1-0716-2974-1_12" data-hide-zero-citations="true" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                        </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [8] D. Vijay Anand, Qiang Xu, <strong>JunJie Wee</strong>, Kelin Xia, Tze Chien Sum. Topological Feature Engineering for Machine Learning Based Halide Perovskite Materials Design. <span style="font-style: italic;">npj Computational Materials</span>. (2022). 8(1), 203.
                        </summary><br>
                                <p> 	Accelerated materials development with machine learning (ML) assisted screening and high throughput experimentation for new photovoltaic materials holds the key to addressing our grand energy challenges. Data-driven ML is envisaged as a decisive enabler for new perovskite materials discovery. However, its full potential can be severely curtailed by poorly represented molecular descriptors (or fingerprints). Optimal descriptors are essential for establishing effective mathematical representations of quantitative structure-property relationships. Here we reveal that our persistent functions (PFs) based learning models offer significant accuracy advantages over traditional descriptor based models in organic-inorganic halide perovskite (OIHP) materials design and have similar performance as deep learning models. Our multiscale simplicial complex approach not only provides a more precise representation for OIHP structures and underlying interactions, but also has better transferability to ML models. Our results demonstrate that advanced geometrical and topological invariants are highly efficient feature engineering approaches that can markedly improve the performance of learning models for molecular data analysis. Further, new structure-property relationships can be established between our invariants and bandgaps. {We anticipate that our molecular representations and featurization models will transcend the limitations of conventional approaches and lead to breakthroughs in perovskite materials design and discovery.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1038/s41524-022-00883-8" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/PF-OIHP" target="_blank" rel="noopener">
                                Code
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.21979/N9/CVJWZ9" target="_blank" rel="noopener">
                                Data
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://blogs.ntu.edu.sg/science/2022/10/10/mathematical-ai-methods-supercharge-the-search-for-perovskite-materials/" target="_blank" rel="noopener">
                                News
                                </a>
                            <a><span class="__dimensions_badge_embed__" data-doi="10.1038/s41524-022-00883-8" data-hide-zero-citations="true" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                        </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [7] <strong>Wee JunJie</strong>, Le Hai Khoi. Korenblum constants for various weighted Fock spaces. <span style="font-style: italic;">Complex Variables and Elliptic Equations</span>. (2022), pp. 1-22.
                        </summary><br>
                                <p> We study the Korenblum Maximum Principle for various weighted Fock spaces. The main tool relies on applying special cases of Ramanujan's Master Theorem involving the Gamma function and the Mellin transform of Dirichlet series. It is interesting that with elementary probe functions, we still can obtain closed-form upper bound expressions, in terms of several well-known special functions, for the Korenblum constants of various weighted Fock spaces. For the first time, we obtain upper bounds of the Korenblum constant for the finite and infinite intersections of weighted Fock spaces and proved that the Korenblum Maximum Principle fails for those infinite intersections of weighted Fock spaces. Some open questions are provided.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1080/17476933.2022.2052862" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a><span class="__dimensions_badge_embed__" data-doi="10.1080/17476933.2022.2052862" data-hide-zero-citations="true" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                </div>

                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [6] Ronald Koh Joon Wei, <strong>JunJie Wee</strong>, Valerie Evangelin Laurent and Kelin Xia. Hodge theory-based biomolecular data analysis. (2022). <span style="font-style: italic;">Scientific Reports</span>. 12(1), 1-16.
                        </summary><br>
                                <p> Hodge theory reveals the deep intrinsic relations of differential forms and provides a bridge between differential geometry, algebraic topology, and functional analysis. Here we use Hodge Laplacian and Hodge decomposition models to analyze biomolecular structures. Different from traditional graph-based methods, biomolecular structures are represented as simplicial complexes, which can be viewed as a generalization of graph models to their higher-dimensional counterparts. Hodge Laplacian matrices at different dimensions can be generated from the simplicial complex. The spectral information of these matrices can be used to study intrinsic topological information of biomolecular structures. Essentially, the number (or multiplicity) of k-th dimensional zero eigenvalues is equivalent to the k-th Betti number, i.e., the number of k-th dimensional homology groups. The associated eigenvectors indicate the homological generators, i.e., circles or holes within the molecular-based simplicial complex. Furthermore, Hodge decomposition-based HodgeRank model is used to characterize the folding or compactness of the molecular structures, in particular, the topological associated domain (TAD) in high-throughput chromosome conformation capture (Hi-C) data. Mathematically, molecular structures are represented in simplicial complexes with certain edge flows. The HodgeRank-based average/total inconsistency (AI/TI) is used for the quantitative measurements of the folding or compactness of TADs. This is the first quantitative measurement for TAD regions, as far as we know.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1038/s41598-022-12877-z" target="_blank" rel="noopener">
                                DOI
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/Hodge-Theory" target="_blank" rel="noopener">
                                    Code
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1038/s41598-022-12877-z" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                </div>
                
                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [5] Gong Weikang, <strong>Wee JunJie</strong>, Wu Min-Chun, Sun Xiaohan, Li Chunhua, Xia Kelin. Persistent spectral simplicial complex-based machine learning (PerSpectSC-ML) for chromosomal structural analysis in cellular differentiation. <span style="font-style: italic;">Briefings in Bioinformatics</span>. (2022). Volume 23, Issue 4, bbac168.
                        </summary><br>
                                <p> The three-dimensional (3D) chromosomal structure plays an essential role in all DNA-templated processes, including gene transcription, DNA replication, and other cellular processes. Although developing chromosome conformation capture (3C) methods, such as Hi-C, which can generate chromosomal contact data characterized genome-wide chromosomal structural properties, understanding 3D genomic nature-based on Hi-C data remains lacking. Here, we propose a persistent spectral simplicial complex (PerSpectSC) model to describe Hi-C data for the first time. Specifically, a filtration process is introduced to generate a series of nested simplicial complexes at different scales. For each of these simplicial complexes, its spectral information can be calculated from the corresponding Hodge Laplacian matrix. PerSpectSC model describes the persistence and variation of the spectral information of the nested simplicial complexes during the filtration process. Different from all previous models, our PerSpectSC-based features provide a quantitative global-scale characterization of chromosome structures and topology. Our descriptors can successfully classify cell types and also cellular differentiation stages for all the 24 types of chromosomes simultaneously. In particular, persistent minimum best characterizes cell types and Dim (1) persistent multiplicity best characterizes cellular differentiation. These results demonstrate the great potential of our PerSpectSC-based models in polymeric data analysis.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1093/bib/bbac168" target="_blank" rel="noopener">
                                DOI
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://github.com/Weikang-Gong/PerSpectSC-ML" target="_blank" rel="noopener">
                                    Code
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1093/bib/bbac168" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                </div>

                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [4] <strong>Wee JunJie</strong>, Xia Kelin. Persistent spectral based ensemble learning (PerSpect-EL) for protein-protein binding affinity prediction. <span style="font-style: italic;">Briefings in Bioinformatics</span>. (2022). Volume 23, Issue 2, bbac024.
                        </summary><br>
                                <p> Protein-protein interactions (PPIs) play a significant role in nearly all cellular and biological activities. Data-driven machine learning models have demonstrated great power in PPIs. However, the design of efficient molecular featurization poses a great challenge for all learning models for PPIs. Here we propose persistent spectral (PerSpect) based PPI representation and featurization, and PerSpect based ensemble learning (PerSpect-EL) models for PPI binding affinity prediction, for the first time. In our model, a sequence of Hodge (or combinatorial) Laplacian (HL)  matrices at various different scales are generated from a specially-designed filtration process. PerSpect attributes, which are statistical and combinatorial properties of spectrum information from these HL matrices, are used as features for PPI characterization. Each PerSpect attribute is input into a 1D convolutional neural network (CNN), and these CNN networks are stacked together in our PerSpect based ensemble learning models. We systematically test our model on the two most commonly-used datasets, i.e., SKEMPI and AB-Bind. It has been found that our model can achieve state-of-the-art results and outperform all existing models to the best of our knowledge. 
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1093/bib/bbac024" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a href="./publications/perspect-el.bibtex" class="btn btn-outline-primary btn-page-header" data-filename="./publications/fprc.bibtex">
                                Cite
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/PerSpect-Ensemble-Learning" target="_blank" rel="noopener">
                                    Code
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1093/bib/bbac024" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                </div>

                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [3] <strong>Wee JunJie</strong>, Xia Kelin. Forman persistent Ricci curvature (FPRC) based machine learning for protein-ligand binding affinity prediction. <span style="font-style: italic;">Briefings in Bioinformatics</span> (2021). Volume 22, Issue 6, bbab136.
                        </summary><br>
                                <p> Artificial intelligence (AI) techniques have already been gradually applied to the
                                    entire drug design process, from target discovery, lead discovery, lead optimization and
                                    preclinical development to the final three phases of clinical trials. Currently, one of
                                    the central challenges for AI-based drug design is molecular featurization, which is to
                                    identify or design appropriate molecular descriptors or fingerprints. Efficient and
                                    transferable molecular descriptors are key to the success of all AI-based drug design
                                    models. Here we propose Forman persistent Ricci curvature (FPRC)-based molecular
                                    featurization and feature engineering, for the first time. Molecular structures and
                                    interactions are modeled as simplicial complexes, which are generalization of graphs to
                                    their higher dimensional counterparts. Further, a multiscale representation is achieved
                                    through a filtration process, during which a series of nested simplicial complexes at
                                    different scales are generated. Forman Ricci curvatures (FRCs) are calculated on the
                                    series of simplicial complexes, and the persistence and variation of FRCs during the
                                    filtration process is defined as FPRC. Moreover, persistent attributes, which are
                                    FPRC-based functions and properties, are employed as molecular descriptors, and combined
                                    with machine learning models, in particular, gradient boosting tree (GBT). Our FPRC-GBT
                                    models are extensively trained and tested on three most commonly-used datasets,
                                    including PDBbind-2007, PDBbind-2013 and PDBbind-2016. It has been found that our
                                    results are better than the ones from machine learning models with traditional molecular
                                    descriptors.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1093/bib/bbab136" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a href="./publications/fprc.bibtex" class="btn btn-outline-primary btn-page-header" data-filename="./publications/fprc.bibtex">
                                Cite
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/Forman-Persistent-Ricci-Curvature" target="_blank" rel="noopener">
                                    Code
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1093/bib/bbab136" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                </div>

                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details><summary>
                            [2] <strong>Wee JunJie</strong>, Xia Kelin. Ollivier Persistent Ricci Curvature-Based Machine Learning for the Protein-Ligand Binding Affinity Prediction. <span style="font-style: italic;">Journal of Chemical Information and Modeling</span> (2021). 61(4), 1617-1626.
                            </summary><br>
                                <p> Efficient molecular featurization is one of the major issues for machine learning models
                                    in drug design. Here, we propose a persistent Ricci curvature (PRC), in particular,
                                    Ollivier PRC (OPRC), for the molecular featurization and feature engineering, for the
                                    first time. The filtration process proposed in the persistent homology is employed to
                                    generate a series of nested molecular graphs. Persistence and variation of Ollivier
                                    Ricci curvatures on these nested graphs are defined as OPRC. Moreover, persistent
                                    attributes, which are statistical and combinatorial properties of OPRCs during the
                                    filtration process, are used as molecular descriptors and further combined with machine
                                    learning models, in particular, gradient boosting tree (GBT). Our OPRC-GBT model is used
                                    in the prediction of the protein–ligand binding affinity, which is one of the key steps
                                    in drug design. Based on three of the most commonly used data sets from the
                                    well-established protein–ligand binding databank, that is, PDBbind, we intensively test
                                    our model and compare with existing models. It has been found that our model can achieve
                                    the state-of-the-art results and has advantages over traditional molecular descriptors.
                                </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://arxiv.org/abs/2011.10281" target="_blank" rel="noopener">
                                arXiv
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1021/acs.jcim.0c01415" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a href="./publications/oprc.bibtex" class="btn btn-outline-primary btn-page-header" data-filename="./publications/oprc.bibtex">
                                Cite
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://github.com/ExpectozJJ/Ollivier-Persistent-Ricci-Curvature" target="_blank" rel="noopener">
                                Code
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://researchdata.ntu.edu.sg/dataset.xhtml;jsessionid=75098d0f62c16693fc598ee4d435?persistentId=doi%3A10.21979%2FN9%2FJHXLSG&version=&q=&fileTypeGroupFacet=%22Code%22&fileAccess=&fileSortField=type" target="_blank" rel="noopener">
                                Data
                                </a>
                            <a><span class="__dimensions_badge_embed__" data-doi="10.1021/acs.jcim.0c01415" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>
                </div>


                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details>
                                <summary>
                            [1] <strong>Wee JunJie</strong>, Le Hai Khoi. Korenblum constants for some function spaces. <span style="font-style: italic;">Proc. Amer. Math. Soc.</span> <strong>148</strong> (2020), pp. 1175-1185.
                        </summary>
                            <br>
                            <p>
                                We study the Korenblum Maximum Principle on the weighted Fock spaces and the weighted
                                Bergman spaces with exponential weights. First, we give explicit expressions for the upper
                                bounds of Korenblum constants for the weighted Fock spaces. Then, we obtain upper bounds of
                                such constants for the weighted Bergman spaces.
                                Finally, we show a failure of the Korenblum Maximum Principle for weighted Bergman spaces
                                $A^p_\alpha(\mathbb{D})$ where $p\in (0,1)$ and $\alpha>0$.
                            </p>
                            </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://www.ams.org/journals/proc/2020-148-03/S0002-9939-2019-14778-X/proc14778_AM.pdf" target="_blank" rel="noopener">
                                Preprint
                                </a>
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.1090/proc/14778" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a href="./publications/kcfs.bibtex" class="btn btn-outline-primary btn-page-header" data-filename="./publications/kcfs.bibtex">
                                Cite
                                </a>
                                <a class="btn btn-outline-primary btn-page-header" href="https://mathscinet.ams.org/mathscinet-getitem?mr=4055945" target="_blank" rel="noopener">
                                    MathSciNet
                                    </a>
                                <a><span class="__dimensions_badge_embed__" data-doi="10.1090/proc/14778" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script>
                                </a>
                            </div>
                    </div>

                </div>
                <br>
                <h2>Theses</h2>
                <p style="font-size: 14px">Click the dissertations to view abstract.</p><br>
                <div class="flexcontainer">
                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details>
                            <summary>
                            [2] <strong>Wee JunJie</strong>. (2023). Geometric and Topological AI for Molecular Sciences. Doctoral Thesis. Nanyang Technological University, Singapore. https://hdl.handle.net/10356/165903</summary><br>
                                <p>
                                	Data-driven sciences are widely regarded as the fourth paradigm of sciences that will fundamentally change the society and our daily lives. Indeed, artificial intelligence (AI) models have already revolutionized and transformed various data-intensive industries. Machine learning (ML) and deep learning models have achieved unprecedented extraordinary performance in image, text, audio, video, and network data analysis. This is largely due to the rise in three major advancements, i.e., accumulation of big data, rise in computational power, and design of highly efficient algorithms. In particular, AlphaFold2 made a remarkable achievement for protein-folding problems which heralds a new era for AI-based molecular data analysis for materials, chemistry, and biology. With excitement and opportunities, AI for molecular sciences also comes with challenges. In this dissertation, we will tackle one of the main challenges in AI for molecular sciences which is constructing or designing effective molecular descriptors and fingerprints. Ideally, effective molecular descriptors should preserve the utmost important features while still possessing the ability to capture the intrinsic molecular properties and information that directly dictate molecular functions. In this way, they can be better “understood” by ML models. This has inspired various researchers to apply topological data analysis (TDA) where persistent homology (PH) and its intrinsic topological invariants act as an excellent and robust molecular featurization method to capture and characterize the underlying topological information in biomolecular systems. By extending beyond the capabilities of TDA, we propose geometric and topological AI for molecular sciences. In this dissertation, two novel persistent functions, namely persistent Ricci curvature (PRC) and persistent Dirac operators are developed as new advanced mathematics-based molecular featurization which can build advanced mathematics-based ML models to perform unsupervised and supervised learning in molecular sciences. In biological data, we built Ollivier persistent Ricci curvature and Forman persistent Ricci curvature-based ML models to predict protein-ligand binding affinity values. Also, we constructed a persistent spectral-based ensemble learning model (PerSpect-EL) to capture and characterize protein-protein interactions upon mutational change. Our PerSpect-EL model has outperformed several existing traditional molecular descriptor-based models in protein-protein binding affinity change predictions. In materials data, we designed both PH-based and persistent Forman curvature (PFC)-based ML models to characterize organic-inorganic halide perovskites (OIHPs). Essentially, our PH-based and PFC-based molecular features produced strong discriminating power in classifying 9 types of OIHPs. Both models have also outperformed traditional perovskite descriptors in materials property predictions such as bandgap, dielectric constant, and refractive index.
                                </p>
                                </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://doi.org/10.32657/10356/165903" target="_blank" rel="noopener">
                                DOI
                                </a>
                            <a><span class="__dimensions_badge_embed__" data-doi="10.32657/10356/165903" data-hide-zero-citations="true" data-style="small_rectangle"></span><script async src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script></a>
                            </div>
                    </div>

                    <div class="bibcontainer">
                        <a class="bibItem" href="javascript:void(0)">
                            <details>
                            <summary>
                            [1] <strong>Wee JunJie</strong>. (2019). The Korenblum Maximum Principle for some function spaces. Bachelor Thesis. Nanyang Technological University, Singapore. https://hdl.handle.net/10356/77142</summary><br>
                                <p>
                                We study the Korenblum Maximum Principle on the weighted Fock space $\mathcal{F}^p_\alpha(\mathbb{C})$ and the 
                                weighted Bergman space $A^p_\alpha(\mathbb{D})$ under exponential weights $e^{−\frac{p\alpha}{2}|z|^2}$. 
                                We obtain explicit expressions for the upper bounds of Korenblum constants for the weighted Fock space $\mathcal{F}^p_\alpha(\mathbb{C})$
                                , $p\geq 1$ and $\alpha > 0$. Then, we obtain upper bounds of such constants for the weighted Bergman space $A^p_\alpha(\mathbb{D})$
                                , $p \geq 1$ and $\alpha \geq 0$. We also show a failure of the Korenblum Maximum Principle for weighted Bergman space $A^p_\alpha(\mathbb{D})$
                                , where $p\in (0,1)$, $\alpha > 0$, thus bringing closure of the problem under weighted Bergman space $A^p_\alpha(\mathbb{D})$ where $\alpha > 0$.
                                </p>
                                </details>
                        </a>
                        <div class="artLinks">
                            <a class="btn btn-outline-primary btn-page-header" href="https://hdl.handle.net/10356/77142" target="_blank" rel="noopener">
                                URI
                                </a>
                            <a href="./publications/kmp_thesis.bibtex" class="btn btn-outline-primary btn-page-header" data-filename="./publications/kmp_thesis.bibtex">
                                Cite
                                </a>
                            </div>
                    </div>

                </div>
            </div><br><br>
        </div>
        </div>
    </div>

    <!-- scripts -->
    <script>
        /* When the user clicks on the button, 
        toggle between hiding and showing the dropdown content */
        function myFunction() {
          document.getElementById("myDropdown").classList.toggle("show");
        }
        
        // Close the dropdown if the user clicks outside of it
        window.onclick = function(event) {
          if (!event.target.matches('.dropbtn')) {
            var dropdowns = document.getElementsByClassName("dropdown-content");
            var i;
            for (i = 0; i < dropdowns.length; i++) {
              var openDropdown = dropdowns[i];
              if (openDropdown.classList.contains('show')) {
                openDropdown.classList.remove('show');
              }
            }
          }
        }
        </script>
    <script type="text/x-mathjax-config"> MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "all" } } }); </script>
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({
            "fast-preview": {disabled:true},
            tex2jax: {
              preview: "none",
              inlineMath: [["$","$"],["\\(","\\)"]],
              processEscapes: true
            }
          });
    </script>
    <script>function openAll() {
        var x = document.getElementsByTagName("details");
        var elem = document.getElementById("ec"); 
        var i;
        if ( elem.innerHTML === '+' ){
            for (i = 0; i < x.length; i++) {
                    x[i].setAttribute("open", "true");
            }
            elem.innerHTML="-";
        }
        else{
            if ( elem.innerHTML === '-' ){
                for (i = 0; i < x.length; i++) {
                        x[i].removeAttribute("open");
                }
                elem.innerHTML="+";
            }
        }
    }
    </script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.js"></script>
    <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/latest.js?config=TeX-MML-AM_CHTML' async></script>
    <script src="particles.js"></script>
    <script src="js/app.js"></script>
    <script async="" src="https://www.google-analytics.com/analytics.js"></script>
    <!-- Google Analytics -->
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
      
      ga('create', 'UA-54900133-1', 'auto');
      ga('send', 'pageview');
      </script>
      <!-- End Google Analytics -->
</body>

</html>